cmd/config.go

```diff
@@ -171,6 +174,71 @@ func readEnvConfig() (conf Config, err error) {
 	return conf, nil
 }
 
+// This checks for conflicting options and turns any shortcut options (i.e. duration, iterations,
+// stages) into the proper scheduler configuration
+func buildExecutionConfig(conf Config) (Config, error) {
+	result := conf
+	if conf.Duration.Valid {
+		if conf.Iterations.Valid {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both duration and iterations is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Stages != nil {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both duration and stages is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Execution != nil {
+			return result, errors.New("specifying both duration and execution is not supported")
+		}
+
+		ds := scheduler.NewConstantLoopingVUsConfig(lib.DefaultSchedulerName)
+		ds.VUs = conf.VUs
+		ds.Duration = conf.Duration
+		result.Execution = scheduler.ConfigMap{lib.DefaultSchedulerName: ds}
+	} else if conf.Stages != nil {
+		if conf.Iterations.Valid {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both iterations and stages is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Execution != nil {
+			return conf, errors.New("specifying both stages and execution is not supported")
+		}
+
+		ds := scheduler.NewVariableLoopingVUsConfig(lib.DefaultSchedulerName)
+		ds.StartVUs = conf.VUs
+		for _, s := range conf.Stages {
+			if s.Duration.Valid {
+				ds.Stages = append(ds.Stages, scheduler.Stage{Duration: s.Duration, Target: s.Target})
+			}
+		}
+		result.Execution = scheduler.ConfigMap{lib.DefaultSchedulerName: ds}
+	} else if conf.Iterations.Valid || conf.Execution == nil {
+		// Either shared iterations were explicitly specified via the shortcut option, or no execution
+		// parameters were specified in any way, which will run the default 1 iteration in 1 VU
+		if conf.Iterations.Valid && conf.Execution != nil {
+			return conf, errors.New("specifying both iterations and execution is not supported")
+		}
+
+		ds := scheduler.NewSharedIterationsConfig(lib.DefaultSchedulerName)
+		ds.VUs = conf.VUs
+		if conf.Iterations.Valid { // TODO: fix where the default iterations value is set... sigh...
+			ds.Iterations = conf.Iterations
+		}
+
+		result.Execution = scheduler.ConfigMap{lib.DefaultSchedulerName: ds}
+	}
+
+	//TODO: validate the config; questions:
+	// - separately validate the duration, iterations and stages for better error messages?
+	// - or reuse the execution validation somehow, at the end? or something mixed?
+	// - here or in getConsolidatedConfig() or somewhere else?
```

Comment:
I would say validate result.Execution and have separate validation in case we are in one of the upper `if`s which than will skip this iteration. So just an `else` here with the validation for the `execution`. 

We should not make non valid configuration from valid cli flags so it should be okay ;)

Reply:
"Valid" CLI flags are a bit of a problem, mostly because the numbers and duration values are signed. So unfortunately those are currently allowed and k6 works with them, albeit strangely: `k6 run --iterations=-1 script.js` and `k6 run --duration=-1s`

Reply:
maybe we should make the negative one illegal ? and check for them above? Do they serve any actual usecase that we can't provide with a more .... none hacky option

Reply:
I think they currently accidentally serve the "infinite duration" use case... Which is also done by explicitly setting the duration to `0`... So yeah, making them "illegal" is part of the reason for the `Validate()` functions in all of the configs, which will currently emit only warnings, but will emit errors and prevent the script execution in the next version. This `TODO` here is just because I still haven't decided what is the best way to inject that validation is, i.e. how to balance the code quality and user-friendliness of the error messages... But it can't be done now anyway, so it will have to be resolved in the next PR, where we can break stuff.